app/controllers/comments_controller.rb

```diff
@@ -1,2 +1,24 @@
 class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = Comment.new(comment_params)
+
+    if @comment.save
+      redirect_to post_path(@comment.post)
+    else
+      render :new
+    end
```

Comment:
- [ ] An error now occurs after saving a comment and attempting to redirect to the `user_post_path` and this is because `post_path` is not defined. This error was not there before.
![image](https://user-images.githubusercontent.com/46839548/232242541-f189a1e2-4d48-4f69-9790-e861a26c9efc.png)

The below code should help you fix this problem 👇 
```ruby
if @comment.save
   redirect_to user_post_path(@comment.post.author, @comment.post)
else
  render :new
end
```

Reply:
Thanks. updated